frontend/model/contracts/chatroom.js

```diff
@@ -129,17 +129,24 @@ function createNotificationData (
 }
 
 function emitMessageEvent ({ type, contractID, hash, state }: {
-  type: string, contractID: string, hash: string, state: Object
+  type: string,
+  contractID: string,
+  hash: string,
+  state?: Object
 }): void {
-  for (let i = state.messages.length - 1; i >= 0; i--) {
-    if (state.messages[i].id === hash) {
-      sbp('okTurtles.events/emit', `${CHATROOM_MESSAGE_ACTION}-${contractID}`, {
-        type,
-        data: { message: state.messages[i] }
-      })
-      break
+  let data = {}
+  if (type === MESSAGE_ACTION_TYPES.ADD_MESSAGE || type === MESSAGE_ACTION_TYPES.EDIT_MESSAGE) {
+    const messagesCount = state?.messages.length || 0
+    for (let i = messagesCount - 1; i >= 0; i--) {
+      if (state?.messages[i].id === hash) {
+        data = { message: { ...state?.messages[i] } }
+        break
+      }
     }
+  } else if (type === MESSAGE_ACTION_TYPES.DELETE_MESSAGE) {
+    data = { id: hash }
```

Comment:
This is confusing me, it suggests that there are two different shapes to the `data` object.

Sometimes `data` contains the `id` directly inside of it, as above: `{ id: hash }`

Other times the `id` is stored inside another object, as on line 142: `{ message: { id, ... } }`

Won't think cause confusion and potential bugs in the rest of the code? Should perhaps this `data` be made more consistent, so that `id` is in the same place in both cases?

Reply:
Got it. I just wanted to send only necessary data especially only `id`, however I needed to keep consistency of the data type. I agree with it.

But can we skip this in this PR? Because we are not sending those data anymore.
You can see what I am saying if you review PR #1274. 

> sbp('okTurtles.events/emit', `${CHATROOM_MESSAGE_ACTION}-${contractID}`, { hash })

I send only `hash` and no other data in the next  PR.
But anyway, I understand your point and will keep it in my mind.

Reply:
> But can we skip this in this PR? Because we are not sending those data anymore.

OK, sure 👍